builder/repo_control.go

```diff
@@ -305,19 +306,36 @@ func (b *Builder) ListRepos() error {
 	if err != nil {
 		return errors.Wrap(err, "unable to load DNF configration, try initializing workspace")
 	}
-
+	b.repoInfo = make(map[string]repoInfo)
 	for _, s := range DNFConf.Sections() {
+		var repoInfo repoInfo
 		name := s.Name()
 		if name == "" {
 			continue
 		}
 
-		url := s.Key("baseurl").Value()
-		if url == "" {
+		u := s.Key("baseurl").Value()
+		if u == "" {
 			continue
 		}
 
-		fmt.Printf("%s\t%s\n", name, url)
+		ur, err := url.Parse(u)
+		if err != nil {
+			return nil
+		}
+		// if file uri scheme, the repo can be accessed offline
+		if ur.Scheme == "file" {
+			ur.Scheme = ""
+			repoInfo.repoURL = ur.String()
+			repoInfo.repoOnline = false
+			b.repoInfo[name] = repoInfo
+		} else {
+			repoInfo.repoURL = u
+			repoInfo.repoOnline = true
+			b.repoInfo[name] = repoInfo
+		}
+
+		fmt.Printf("%s\t%s\n", name, ur.String())
```

Comment:
This logic should be in `AddRepo()` instead of in `ListRepos()`. Hence we can have it at just one place.
E.g.
```
	pURL, err := _url.Parse(url)
	if err != nil {
		return err
	}

	if pURL.Scheme == "" { // if url scheme is empty, explicitly set it to "file"
		pURL.Scheme = "file"
	}

	if b.Repos == nil {
		b.Repos = make(map[string]repoInfo)
	}
	var repos repoInfo
	repos.url = pURL.String()
	repos.urlScheme = pURL.Scheme
	b.Repos[name] = repos

	fmt.Printf("Adding %s at %s\n", name, repos.url)
```

Reply:
I didnt understand how we can add this logic to AddRepo, AddRepo is a separate cmd which will exit after adding the repo. We wont iterate over all the repos.

Reply:
Actually you are correct. Only this should be here:
```
	pURL, err := _url.Parse(url)
	if err != nil {
		return err
	}

	if pURL.Scheme == "" { // if url scheme is empty, explicitly set it to "file"
		pURL.Scheme = "file"
	}
	fmt.Printf("Adding %s at %s\n", name, pURL.String())

Creation of the repoInfo map can be in ListRepo like you already have. Coming to think of it, it will also handle situations where the user has provided their own DNFConf which I wasn't thinking about.
```